sdk/nodejs/runtime/invoke.ts

```diff
@@ -67,18 +68,37 @@ const providerproto = require("../proto/provider_pb.js");
  */
 export function invoke(tok: string, props: Inputs, opts: InvokeOptions = {}): Promise<any> {
     if (opts.async) {
-        // Use specifically requested async invoking.  Respect that.
+        // User specifically requested async invoking.  Respect that.
         return invokeAsync(tok, props, opts);
     }
 
+    const config = new Config("pulumi");
+    const noSyncCalls = config.getBoolean("noSyncCalls");
+    if (noSyncCalls) {
+        // User globally disabled sync invokes.
+        return invokeAsync(tok, props, opts);
+    }
+
+    const syncResult = invokeSync(tok, props, opts);
+
+    // Wrap the synchronous value in a Promise view as well so that consumers can treat it
+    // either as the real value or something they can use as a Promise.
+    return createLiftedPromise(syncResult);
```

Comment:
instead of all the invokeSync codepaths having to call createLiftedPromise, i 'lifted' (no pun intended) all of that to this single location.